### PR TITLE
Fixed some frontend tests causing unhandled rejection errors

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Fixed a few tests that were causing unhandled rejection errors

## Related Tickets & Documents

#11626

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how accessibility is impacted and tested. For more info, check out the [Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Michael Scott from The Office saying "Clean up in aisle five"

### CODE REVIEW

File: testSetup.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
 /* eslint-env node */
 import 'jest-axe/extend-expect';
 import './app/assets/javascripts/lib/xss';
+
+process.on('unhandledRejection', (error) => {
```
Comment (nickytonline) [inline testSetup.js:5]: This prevents the these errors from entering the testing codebase.

File: app/javascript/onboarding/components/__tests__/FollowUsers.test.jsx [hunk 2]
```diff
@@ -81,6 +81,8 @@ describe('FollowUsers', () => {
   });
 
   it('should render the correct navigation button on first load', () => {
+    fetch.mockResponseOnce(fakeUsersResponse);
```
Comment (nickytonline) [inline app/javascript/onboarding/components/__tests__/FollowUsers.test.jsx:84]: No response was being faked, hence the unhandled rejection.

File: app/javascript/chat/__tests__/requestManager.test.jsx [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
+    window.getCsrfToken = async () => csrfToken;
+
+    fetch.mockResponse(
+      JSON.stringify({
+        result: { user_joining_requests: [], channel_joining_memberships: [] },
+      }),
+    );
+  });
+
+  it('should have no a11y violations', async () => {
```
Comment (nickytonline) [inline app/javascript/chat/__tests__/requestManager.test.jsx:33]: Was missing a11y tests.

File: app/javascript/chat/__tests__/requestManager.test.jsx [hunk 4]
```diff
@@ -1,17 +1,48 @@
...
+
+// TODO: There needs to be some better tests in here in regards to different data: empty, some data etc.
 describe('<RequestManager />', () => {
+  beforeEach(() => {
+    const csrfToken = 'this-is-a-csrf-token';
+
+    window.fetch = fetch;
+    window.getCsrfToken = async () => csrfToken;
+
+    fetch.mockResponse(
```
Comment (nickytonline) [inline app/javascript/chat/__tests__/requestManager.test.jsx:26]: No response was being faked, hence the unhandled rejection.

File: app/javascript/chat/__tests__/requestManager.test.jsx [hunk 5]
```diff
@@ -1,17 +1,48 @@
...
+  const data = [
+    {
+      resource: {},
+    },
+  ];
 
+  return data;
+}
+
+// TODO: There needs to be some better tests in here in regards to different data: empty, some data etc.
```
Comment (nickytonline) [inline app/javascript/chat/__tests__/requestManager.test.jsx:18]: Added this for future work as it's not related to this PR.

File: app/javascript/article-form/components/__tests__/ImageUploader.test.jsx [hunk 6]
```diff
@@ -106,9 +106,10 @@ describe('<ImageUploader />', () => {
       },
     });
 
-    expect(await findByText(/uploading.../i)).toBeDefined();
+    expect(await findByText(/uploading.../i)).not.toBeNull();
 
-    waitForElementToBeRemoved(() => queryByText(/uploading.../i));
+    // Upload is finished, so the messsage has disappeared.
+    expect(queryByText(/uploading.../i)).toBeNull();
```
Comment (nickytonline) [inline app/javascript/article-form/components/__tests__/ImageUploader.test.jsx:112]: At this point the `uploading...` message is gone, so we need to use [queryByText](https://testing-library.com/docs/dom-testing-library/api-queries/#queryby) instead of findByText since it is already removed.

File: app/javascript/chat/__tests__/requestManager.test.jsx (hunk 3 above)
Comment (nickytonline) [reply app/javascript/chat/__tests__/requestManager.test.jsx:33]: This could be a separate PR. Happy to put it in a separate PR if need be.

**Review by Ridhwana — APPROVED** [review]

**Review by rhymes — APPROVED** [review]
Great catch!
